rust-runtime/aws-smithy-client/src/lib.rs

```diff
@@ -226,7 +228,47 @@ where
             .layer(DispatchLayer::new())
             .service(connector);
 
-        check_send_sync(svc).ready().await?.call(input).await
+        // send_operation records the full request-response lifecycle.
+        // NOTE: For operations that stream output, only the setup is captured in this span.
+        let span = debug_span!(
+            "send_operation",
+            operation = field::Empty,
+            service = field::Empty,
+            status = field::Empty,
+            message = field::Empty
+        );
+        let (mut req, parts) = op.into_request_response();
+        if let Some(metadata) = &parts.metadata {
+            span.record("operation", &metadata.name());
+            span.record("service", &metadata.service());
+            // This will clone two `Cow::<&'static str>::Borrow`s in the vast majority of cases
+            req.properties_mut().insert(metadata.clone());
+        }
```

Comment:
Is it worth adding a method to operation that exposes this metadata?

Reply:
@rcoh was arguing we should remove metadata from the operation entirely and just have something in the property bag, but that was looking to be a pretty sizable refactor that I didn't want to take on here.

I can take a look and see if it's easy to just borrow metadata from the operation here to eliminate the destructuring/restructuring.

Reply:
It seems promising at first, but then ends up not working out since calling `op.metadata()` to borrow the metadata locks the entire operation struct so that a property can't be inserted.